dsgrid/registry/registry_auto_updater.py

```diff
@@ -0,0 +1,306 @@
+import logging
+from typing import Any, Optional
+
+from dsgrid.config.mapping_tables import MappingTableConfig, MappingTableModel
+from dsgrid.config.dataset_config import DatasetConfig, DatasetConfigModel
+from dsgrid.config.dimension_config import DimensionConfig
+from dsgrid.config.project_config import ProjectConfig
+from dsgrid.exceptions import DSGInvalidParameter
+from dsgrid.registry.registration_context import RegistrationContext
+from dsgrid.registry.common import (
+    ConfigKey,
+    RegistryType,
+    VersionUpdateType,
+)
+from dsgrid.registry.registry_manager import RegistryManager
+
+
+logger = logging.getLogger(__name__)
+
+
+class RegistryAutoUpdater:
+    """Performs auto-updates on the registry."""
```

Comment:
I'm not sure what an "auto-update" is. (Hopefully reading through the file will enlighten me.)